includes/class-fs-garbage-collector.php

```diff
@@ -0,0 +1,359 @@
+                if ( ! is_object( $product_data ) ) {
+                    continue;
+                }
+
+                if (
+                    empty( $product_data->last_load_timestamp ) ||
+                    ! is_numeric( $product_data->last_load_timestamp )
+                ) {
+                    // Set to the current time so that if the product is no longer used, its data will be deleted after 4 weeks.
+                    $product_data->last_load_timestamp = time();
+
+                    $has_updated_option = true;
+
+                    continue;
+                }
+
+                if ( $this->is_product_active( $slug, $product_data ) ) {
+                    continue;
+                }
+
+                $is_addon = FS_Plugin::is_valid_id( $product_data->parent_plugin_id );
+
+                if ( ! $is_addon ) {
+                    $products_to_delete[] = $product_data;
+                } else {
+                    /**
+                     * If add-on, add to the beginning of the array so that add-ons are removed before their parent. This is to prevent an unexpected issue when an add-on exists but its parent was already removed.
+                     */
+                    array_unshift( $products_to_delete, $product_data );
+                }
+            }
+
+            if ( $has_updated_option ) {
+                $_accounts->store();
+            }
+
+            return $products_to_delete;
+        }
+
+        /**
+         * @param string $slug
+         * @param object $product_data
+         *
+         * @return bool
+         */
+        private function is_product_active( $slug, $product_data ) {
+            $products_to_skip_by_type_and_slug = $this->get_products_to_skip_by_type_and_slug();
+
+            if ( isset( $products_to_skip_by_type_and_slug[ $product_data->type ][ $slug ] ) ) {
+                return true;
+            }
+
+            if ( $product_data->last_load_timestamp > ( time() - ( WP_FS__TIME_WEEK_IN_SEC * 4 ) ) ) {
+                // Last activation was within the last 4 weeks.
+                return true;
+            }
+
+            return false;
+        }
+
+        /**
+         * @param FS_Options $_accounts
+         * @param array[]    $products_to_delete
+         *
+         * @return bool
+         */
+        private function delete_products( $_accounts, $products_to_delete ) {
+            $products_option_names_by_type       = $this->get_product_option_names_by_type();
+            $products_options_by_type            = array();
+            $has_loaded_products_options_by_type = array_fill_keys( array_keys( $products_to_delete ), false );
+
+            $has_updated_option = false;
+
+            foreach( $products_to_delete as $product_type => $products ) {
+                foreach( $products as $product ) {
+                    $slug = $product->slug;
+
+                    if ( ! $has_loaded_products_options_by_type[ $product_type ] ) {

```

Comment:
@fajardoleo continuing from the above two suggestion, here you can do

```php
if ( ! isset($products_options_by_type[ $product_type ])) {
  $products_options_by_type[ $product_type ] = $this->load_options($_accounts, $this->get_product_option_names( $product_type );
}
```

So:
1. Rename/Refactor `get_product_option_names_by_type` to `get_product_option_names`.
2. The function will take the `$product_type` as param and would return the relevant options.